File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/var/pde/HMSampledSpatialVariable.java
package ru.nstu.isma.hsm.var.pde;

import ru.nstu.isma.hsm.var.HMConst;

import java.io.Serializable;

/**
 * Bessonov Alex.
 * Date: 04.12.13 Time: 0:17
 */
public class HMSampledSpatialVariable extends HMSpatialVariable implements Serializable {

    // апроксимировать по шагу или количеству точек
    protected ApproximateType type;

    // в зависимости от типа - шаг или количество точек
    protected HMConst apxVal;

    public Double getStepSize() {
        Double out = null;
        if (type == ApproximateType.BY_NUMBER_OF_PIECES) {
            out = (valTo.getValue() - valFrom.getValue()) / apxVal.getValue();
        } else if (type == ApproximateType.BY_STEP) {
           out = apxVal.getValue();
        }
        return out;
    }

    public Integer getPointsCount() {
        Double out = null;
        if (type == ApproximateType.BY_NUMBER_OF_PIECES) {
            out = apxVal.getValue();
        } else if (type == ApproximateType.BY_STEP) {
            out = (valTo.getValue() - valFrom.getValue()) / apxVal.getValue();
        }
        out = Math.ceil(out);
        return out.intValue();
    }

    public ApproximateType getType() {
        return type;
    }

    public void setType(ApproximateType type) {
        this.type = type;
    }

    public HMConst getApxVal() {
        return apxVal;
    }

    public void setApxVal(HMConst apxVal) {
        this.apxVal = apxVal;
    }

    public enum ApproximateType implements Serializable {
        BY_STEP, BY_NUMBER_OF_PIECES
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/var/pde/HMSampledSpatialVariable.kt
package ru.nstu.isma.hsm.`var`.pde

import ru.nstu.isma.hsm.`var`.HMConst
import java.io.Serializable

/**
 * Bessonov Alex.
 * Date: 04.12.13 Time: 0:17
 */
class HMSampledSpatialVariable : HMSpatialVariable(), Serializable {
    // апроксимировать по шагу или количеству точек
    var type: ApproximateType? = null

    // в зависимости от типа - шаг или количество точек
    var apxVal: HMConst? = null
    val stepSize: Double?
        get() {
            var out: Double? = null
            if (type == ApproximateType.BY_NUMBER_OF_PIECES) {
                out = (valTo!!.value!! - valFrom!!.value!!) / apxVal!!.value!!
            } else if (type == ApproximateType.BY_STEP) {
                out = apxVal!!.value
            }
            return out
        }
    val pointsCount: Int
        get() {
            var out: Double? = null
            if (type == ApproximateType.BY_NUMBER_OF_PIECES) {
                out = apxVal!!.value
            } else if (type == ApproximateType.BY_STEP) {
                out = (valTo!!.value!! - valFrom!!.value!!) / apxVal!!.value!!
            }
            out = Math.ceil(out!!)
            return out.toInt()
        }

    enum class ApproximateType : Serializable {
        BY_STEP, BY_NUMBER_OF_PIECES
    }
}